core/templates/pages/exploration-editor-page/history-tab/history-tab.component.ts

```diff
@@ -61,34 +62,24 @@ angular.module('oppia').component('historyTab', {
         if (ctrl.selectedVersionsArray.length === 2) {
           ctrl.changeCompareVersion();
           ctrl.hideHistoryGraph = false;
-          ctrl.compareVersionsButtonIsHidden = true;
         }
       };
       // Changes the checkbox selection and provides an appropriate user
       // prompt.
-      ctrl.changeSelectedVersions = function(evt, versionNumber) {
-        var checkbox = evt.target;
-        var selectedVersionsArrayPos = ctrl.selectedVersionsArray.indexOf(
-          versionNumber);
-        if (checkbox.checked && selectedVersionsArrayPos === -1) {
-          ctrl.selectedVersionsArray.push(versionNumber);
+      ctrl.changeSelectedVersions = function(snapshot, item) {
+        if (item === 1 && snapshot && !ctrl.selectedVersionsArray.includes(
+          snapshot.versionNumber)) {
+          ctrl.selectedVersionsArray[0] = snapshot.versionNumber;
         }
-        if (!checkbox.checked && selectedVersionsArrayPos !== -1) {
-          ctrl.selectedVersionsArray.splice(selectedVersionsArrayPos, 1);
+        if (item === 2 && snapshot && !ctrl.selectedVersionsArray.includes(
+          snapshot.versionNumber)) {
+          ctrl.selectedVersionsArray[1] = snapshot.versionNumber;
```

Comment:
Is it guaranteed that the length will be at least 2 here? (same for above)
Maybe check that as well?

Reply:
It's two dropdowns, and the values corresponding to them. Hence it'll always be equal to 2.